### PR TITLE
adds support for custom keyFunc for signing and verification keys

### PR DESCRIPTION
**Pull request by bhoriuchi**
addresses https://github.com/go-chi/jwtauth/issues/27 by allowing a custom keyFunc to be passed as the signKey and/or verifyKey

### CODE REVIEW

**Comment by pkieltyka** [comment]
the PR still needs some work.
1. has a few typos (comments)
2. do we really need to use reflection to pull this off?

**Comment by bhoriuchi** [comment]
@pkieltyka can probably accomplish this w/o reflect using .(func....). Will look into it and the comment typos when I get some time

File: jwtauth.go [hunk 1]
```diff
@@ -147,11 +148,22 @@ func (ja *JWTAuth) Decode(tokenString string) (t *jwt.Token, err error) {
 	return
 }
 
+// if the interface is itself a function call that function with the token
+// and return its results. this allows for use cases like jwks without
+// changing the api
+func keyFuncOrValue(i interface{}, t *jwt.Token) (interface{}, error) {
```
Comment (rbUUbr) [inline jwtauth.go]: IMHO, this case is not about reflection usage, I think, we can pass function explicitly, and make type for this function

**Comment by pkieltyka** [comment]
underlying jwt library has changed, see HEAD

**Pull request closed without merging**